geopyspark/geotrellis/catalog.py

```diff
@@ -253,53 +83,68 @@ def read_value(layer_type,
-        options = options or kwargs or {}
+        store = AttributeStore.build(uri)
+        _cached_stores[uri] = store
 
-        if zdt:
-            zdt = zdt.astimezone(pytz.utc).isoformat()
-        else:
-            zdt = ''
+    reader = ValueReader(uri, layer_name, layer_zoom, store)
+    return reader.read(col, row, zdt)
 
-        if uri not in _mapped_cached:
-            _construct_catalog(get_spark_context(), uri, options)
 
-        cached = _mapped_cached[uri]
+class ValueReader(object):
+    """GeoTrellis catalog indivual value reader.
+    Suitable for use in TMS service because it does not have Spark overhead.
+    """
 
-        key = map_key_input(LayerType(layer_type).value, True)
+    def __init__(self, uri, layer_name, zoom=None, store=None):
+        self.layer_name = layer_name
+        self.zoom = zoom
+        self.store = store or AttributeStore(uri)
+        pysc = get_spark_context()
+        scala_store = self.store.wrapper.attributeStore()
+        ValueReaderWrapper = pysc._gateway.jvm.geopyspark.geotrellis.io.ValueReaderWrapper
+        self.wrapper = ValueReaderWrapper(scala_store, uri)
 
-        values = cached.value_reader.readTile(key,
-                                              layer_name,
-                                              layer_zoom,
-                                              col,
-                                              row,
-                                              zdt)
+    def read(self, col, row, zdt=None, zoom=None):
+        """Reads a single ``Tile`` from a GeoTrellis catalog.
+           When requesting a tile that does not exist, ``None`` will be returned.
 
-        return multibandtile_decoder(values)
+        Args:
 
+            col (int): The col number of the tile within the layout. Cols run east to west.
+            row (int): The row number of the tile within the layout. Row run north to south.
+            zdt (``datetime.datetime``): The time stamp of the tile if the data is spatial-temporal.
+                This is represented as a ``datetime.datetime.`` instance.  The default value is,
+                ``None``. If ``None``, then only the spatial area will be queried.
+            zoom (int, optional): The zoom level of the layer that is to be read.
+                Defaults to ``self.zoom``
 
-def query(layer_type,
-          uri,
+        Returns:
+            :class:`~geopyspark.geotrellis.Tile`
+        """
+
+        zoom = zoom or self.zoom or 0
+        zoom = zoom and int(zoom)
+        zdt = zdt and zdt.astimezone(pytz.utc).isoformat()
+        value = self.wrapper.readTile(self.layer_name, zoom, col, row, zdt)
+        return value and multibandtile_decoder(value)
+
+
+def query(uri,
           layer_name,
           layer_zoom=None,
           query_geom=None,
           time_intervals=None,
           query_proj=None,
-          options=None,
           num_partitions=None,
-          **kwargs):
+          attribute_store=None):
```

Comment:
`AttributeStore` and `LayerReader` must not always share the same storage mechanism, although thats true by default. You could be pulling tiles from S3 but have the attributes stored in Accumulo/PostgreSQL/filesystem. 

Being able to decouple the attribute store instance used and the layer URI is whats required to take advantage of that feature. Having an optional argument for it is minimal way to enable that use case.